packages/plugin-nft-generation/src/provider/wallet/walletSolana.ts

```diff
@@ -0,0 +1,231 @@
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import {
+    generateSigner,
+    keypairIdentity,
+    percentAmount,
+    publicKey,
+    sol,
+    TransactionBuilder,
+    Umi,
+} from "@metaplex-foundation/umi";
+import { getExplorerLink } from "@solana-developers/helpers";
+import { transferSol } from "@metaplex-foundation/mpl-toolbox";
+import bs58 from "bs58";
+import { elizaLogger } from "@ai16z/eliza";
+
+export class WalletSolana {
+    private cache: NodeCache;
+    private umi: Umi;
+    private cluster: Cluster;
+
+    constructor(
+        private walletPublicKey: PublicKey,
+        private walletPrivateKeyKey: string,
+        private connection?: Connection
+    ) {
+        this.cache = new NodeCache({ stdTTL: 300 }); // Cache TTL set to 5 minutes
+
+        if (!connection) {
+            this.cluster = (process.env.SOLANA_CLUSTER as Cluster) || "devnet";
+            this.connection = new Connection(clusterApiUrl(this.cluster), {
+                commitment: "finalized",
+            });
+        }
+        const umi = createUmi(this.connection.rpcEndpoint);
+        umi.use(mplTokenMetadata());
+        const umiUser = umi.eddsa.createKeypairFromSecretKey(
+            this.privateKeyUint8Array
+        );
+        umi.use(keypairIdentity(umiUser));
+        this.umi = umi;
+    }
+
+    async getBalance() {
+        let balance = await this.connection.getBalance(this.walletPublicKey);
+        return {
+            value: balance,
+            formater: `${balance / LAMPORTS_PER_SOL} SOL`,
+        };
+    }
+
+    get privateKeyUint8Array() {
+        return bs58.decode(this.walletPrivateKeyKey);
+    }
+
+    async createCollection({
+        name,
+        symbol,
+        adminPublicKey,
+        uri,
+        fee,
+    }: {
+        name: string;
+        symbol: string;
+        adminPublicKey: string;
+        uri: string;
+        fee: number;
+    }): Promise<any> {
+        const collectionMint = generateSigner(this.umi);
+        elizaLogger.log("collectionMint", collectionMint, percentAmount(5));
+        let transaction = new TransactionBuilder();
+        const info = {
+            name,
+            symbol,
+            uri,
+        };
+        console.log(`Metadata uploaded: ${uri}`);
+        transaction = transaction.add(
+            createNft(this.umi, {
+                ...info,
+                mint: collectionMint,
+                sellerFeeBasisPoints: percentAmount(fee),
+                isCollection: true,
+            })
+        );
+
+        transaction = transaction.add(
+            updateV1(this.umi, {
+                mint: collectionMint.publicKey,
+                newUpdateAuthority: publicKey(adminPublicKey), // updateAuthority's public key
+            })
+        );
+
+        await transaction.sendAndConfirm(this.umi, {
+            confirm: {},
+        });
+
+        // await sleep(5000);
+        // const createdCollectionNft = await fetchDigitalAsset(
```

Comment:
What was the commented out code for? Any reason to keep it?

Reply:
fetchDigitalAsset verifies NFT asset confirmation on blockchain.
Since we're using {commitment: "finalized"} in commitmentOrConfig for new Connection(), I think this part can be removed.